sfs/time/drivingfunction.py

```diff
@@ -1,64 +1,198 @@
+    delays = inner1d(n, x0) / c
+    weights = 2 * g0 * inner1d(n, n0)
+    return delays, weights
+
+
+def wfs_25d_point(x0, n0, xs, xref=[0, 0, 0], c=None):
+    r"""Point source by 2.5-dimensional WFS.
+
+    Parameters
+    ----------
+    x0 : (N, 3) array_like
+        Sequence of secondary source positions.
+    n0 : (N, 3) array_like
+        Sequence of secondary source orientations.
+    xs : (3,) array_like
+        Virtual source position.
+    xref : (3,) array_like, optional
+        Reference position
+    c : float, optional
+        Speed of sound
+
+    Returns
+    -------
+    delays : (N,) numpy.ndarray
+        Delays of secondary sources in seconds.
+    weights: (N,) numpy.ndarray
+        Weights of secondary sources.
+
+    Notes
+    -----
+    2.5D correction factor
 
+    .. math::
 
-def delayline(sig, delay, weight):
-    """Delayline with weights, no fractional delay yet.
+         g_0 = \sqrt{2 \pi |x_\mathrm{ref} - x_0|}
 
-    Delay in samples.
-    sig (m x n) where columns n are channels and rows m the corresponding
-    signals.
-    Weight is array with weighting for each channel.
+
+    d using a point source as source model
+
+    .. math::
+
+         d_{2.5D}(x_0,t) = h(t)
+         \frac{g_0  \scalarprod{(x_0 - x_s)}{n_0}}
+         {2\pi |x_0 - x_s|^{3/2}}
+         \dirac{t - \frac{|x_0 - x_s|}{c}}
+
+    with wfs(2.5D) prefilter h(t), which is not implemented yet.
+
+    References
+    ----------
+    See http://sfstoolbox.org/en/latest/#equation-d.wfs.ps.2.5D
 
     """
-    delay = np.rint(delay).astype(int)  # no fractional delay, samples
-    extention = np.max(delay)  # avoid cutting signal
-    channels = len(weight)  # extract no. of channels
-    sig_length = sig.ndim
-    out = np.zeros([sig_length+extention, channels])  # create shape
+    if c is None:
+        c = defs.c
+    x0 = util.asarray_of_rows(x0)
+    n0 = util.asarray_of_rows(n0)
+    xs = util.asarray_1d(xs)
+    xref = util.asarray_1d(xref)
+    g0 = np.sqrt(2 * np.pi * np.linalg.norm(xref - x0, axis=1))
+    ds = x0 - xs
+    r = np.linalg.norm(ds, axis=1)
+    delays = r/c
+    weights = g0/(2 * np.pi) * inner1d(ds, n0) / r**(3/2)
+    return delays, weights
+
+
+def driving_signals(delays, weights, signal, fs=None):
+    """Get driving signals per secondary source.
+
+    Returned signals are the delayed and weighted mono input signal (with N
+    samples) per channel (C).
+
+    Parameters
+    ----------
+    delays : (C,) array_like
+        Delay in seconds for each channel, negative values allowed.
+    weights : (C,) array_like
+        Amplitude weighting factor for each channel.
+    signal : (N,) array_like
+        Excitation signal (mono) which gets weighted and delayed.
+    fs: int, optional
+        Sampling frequency in Hertz.
+
+    Returns
+    -------
+    driving_signals : (N, C) numpy.ndarray
+        Driving signals (with N samples) per channel C.
```

Comment:
I think you should extend the parentheses until the end of the line, or remove them completely (which might be even better).